### PR TITLE
add check for user stream exists or not

### PR DESCRIPTION
**Pull request by codenamejason**

### CODE REVIEW

File: packages/hardhat/contracts/StreamFactory.sol [hunk 1]
```diff
@@ -47,6 +55,9 @@ contract StreamFactory is AccessControl, Ownable {
         bool _startsFull,
         IERC20 _gtc
     ) public isPermittedFactoryManager returns (address streamAddress) {
+        User storage user = users[_toAddress];
+        require(user.hasStream == false, "User already has a stream!");
```
Comment (damianmarti) [inline packages/hardhat/contracts/StreamFactory.sol:59]: I think that maybe you can check this without the new users mapping, just checking userStreams[_toAddress] != address(0).

**Comment by damianmarti** [comment]
I think that is need a check in the addStreamForUser function too.

File: packages/hardhat/contracts/StreamFactory.sol (hunk 1 above)
Comment (codenamejason) [reply packages/hardhat/contracts/StreamFactory.sol:59]: how does that check for an existing stream for that user, that is the zero address?

Comment (damianmarti) [reply packages/hardhat/contracts/StreamFactory.sol:59]: The mapping to address is initialized with address(0), so you can check if the userStream for a user address was not set comparing the value with address(0)

**Pull request merged**